src/Spice86.Shared/Emulator/VM/Breakpoint/Serializable/SerializedBreakpoint.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace Spice86.Shared.Emulator.VM.Breakpoint.Serializable;
+
+using System.Text.Json.Serialization;
+
+/// <summary>
+/// Base class for serializable breakpoint data.
+/// </summary>
+[JsonPolymorphic(TypeDiscriminatorPropertyName = "$type")]
+[JsonDerivedType(typeof(SerializedBreakpoint), typeDiscriminator: "breakpoint")]
+[JsonDerivedType(typeof(SerializedBreakpointRange), typeDiscriminator: "range")]
+public record SerializedBreakpoint {
+    /// <summary>
+    /// Gets the format version.
+    /// </summary>
+    public int Version { get; init; } = 1;
```

Comment:
I think its better to put the version onto the container (ProgramSerializableBreakpoints). I see no point to do that per breakpoint.